lib/request_headers_logger/railtie.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RequestHeadersLogger
+  # The Railtie triggering a setup from RAILs to make it configurable
+  class Railtie < ::Rails::Railtie
+    config.before_configuration do
+      config.request_headers_logger = ::ActiveSupport::OrderedOptions.new
+      config.request_headers_logger.sidekiq = ::ActiveSupport::OrderedOptions.new
+      config.request_headers_logger.sidekiq.proxy_headers = true
+      config.request_headers_logger.sidekiq.use_rails_logger = true
+      config.request_headers_logger.sidekiq.log_with_request_id = true
+      config.request_headers_logger.sidekiq.jobs_logger = if defined?(Sidekiq)
+                                                            ::RequestHeadersLogger::Sidekiq::JobLogger
+                                                          end
+      config.request_headers_logger.sidekiq.exceptions_logger = if defined?(Sidekiq)
+                                                                  ::RequestHeadersLogger::Sidekiq::ExceptionLogger
+                                                                end
+    end
+
+    initializer 'request_headers_logger.sidekiq_configuration' do
+    end
+
+    # rubocop:disable Metrics/BlockLength
+    config.after_initialize do
+      if defined?(Sidekiq)
+        ::Sidekiq::Logging.logger = ::Rails.logger if config.request_headers_logger.sidekiq.use_rails_logger
+
+        if config.request_headers_logger.sidekiq.proxy_headers
```

Comment:
what about spliting the after_initialize to 2 methods `initialize_Sidekiq_Middleware` and `initialize_Sidekiq_loggers` instead of using  "# rubocop:disable Metrics/BlockLength"